src/service/twitterApi.ts

```diff
@@ -0,0 +1,199 @@
+            }
+        } else {
+            return {
+                status: false,
+                result: `💥  Bad request(invalid twitter), please double check your twitter link.`
+            }
+        }
+    } else {
+        return {
+            status: false,
+            result: `💥  Bad request(invalid twitter), please double check your twitter link.`
+        }
+    }
+}
+
+export async function singleTweet(tweetId: string) {
+    // {
+    //     "data": {
+    //       "id": "1067094924124872705",
+    //       "text": "Just getting started with Twitter APIs? Find out what you need in order to build an app. Watch this video! https://t.co/Hg8nkfoizN"
+    //     }
+    // }
+    return roClient.v2.singleTweet(tweetId);
+}
+
+export async function judgeTwitterIdentityByTwitterNum (twNum: string) {
+    const tweet = await singleTweet(twNum);
+    console.log('tweet', tweet)
+    const data = tweet.data;
+    if (data) {
+        const twText = tweet.data.text;    
+        try {
+            const text = twText.replace(/\s+/g, " ")
+            const containFSSupertalk = _.includes(text, `#CrustFreeStorage`);
+            const containCrustSupertalk = _.includes(text.replace(/\s+/g, " "), '#CrustNetwork');
+            if (containFSSupertalk) {
+                if (containCrustSupertalk) {
+                    // like {address} with {protionCode} on the #CrustNetwork via https://discord.gg/WQQHnyKCmn
+                    const addrWithCodeStr = text.substr(twitterContentStart.length);
+                    // like [`cTMeMr6cC2xQwonTwpbSyKGv2VkvxEB836xr63vt8HsDNbF9q`, `protionCode on the #CrustNetwork via https://discord.gg/WQQHnyKCmn`]
+                    const addressSplits = addrWithCodeStr.split(twitterContentPromotionWith);
+                    console.log('addressSplits', addressSplits)
+                    if (addressSplits.length == 2) {
+                        const address = addressSplits[0].trim();
+                        const mainnetAddr = getMainnetAddr(address);
+                        if (mainnetAddr) {
+                            // like ['code', 'via https://discord.gg/WQQHnyKCmn']
+                            const codeSplit = addressSplits[1].split(`on the #CrustNetwork`);
+                            const code = codeSplit[0].trim();
+                            console.log('code', code)
+                            return {
+                                status: true,
+                                address: mainnetAddr,
+                                code
+                            }
+                        } else {
+                            return {
+                                status: false,
+                                result: `💥  Invalid Crust address`
+                            }
+                        }
+                    } else {
+                        return {
+                            status: false,
+                            result: `💥  Wrong content format`
```

Comment:
Make sure xx space with space yy

Reply:
💥   Bad request(bad content format), make sure you tweet like `... {CRUST_ADDRESS} with {PROMO_CODE} ...`